Update engineering handbook README.md with deep freeze (#16550)
fleetdm/fleet: handbook/engineering/README.md

2. The Engineering Manager notifies the QA lead for the product group and the [release ritual DRI](https://fleetdm.com/handbook/engineering#rituals). 
3. The Engineering Manager, QA lead, and [release ritual DRI](#rituals) must all approve the feature work PR before it is unfrozen and merged.

> To have a final clean release test that doesn't get interrupted with more QA requirements the final 24 hours of the freeze will be a "Deep Freeze" where even bugs and doc changes should be held back to allow QA uniterrupted testing. The only exception is a `~release-blocker` or `~unreleased-bug` fix which would then push the release out another day.

Reviewer: ```suggestion
> To allow a stable release test, the final 24 hours before release is a deep freeze when only bugs with the `~release-blocker` or `~unreleased-bug` labels are merged.
```
